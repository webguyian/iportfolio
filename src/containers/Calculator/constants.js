export const add = '+';
export const allClear = 'AC';
export const backspace = 'Backspace';
export const clear = 'C';
export const decimal = '.';
export const divide = '/';
export const divideAlt = '\u00f7';
export const enter = 'Enter';
export const equals = '=';
export const multiply = '*';
export const multiplyAlt = '\u00d7';
export const percent = '%';
export const positiveNegative = '+/-';
export const subtract = '-';
export const zero = '0';

export const controls = {
  add,
  allClear,
  clear,
  decimal,
  divide,
  divideAlt,
  equals,
  multiply,
  multiplyAlt,
  percent,
  positiveNegative,
  subtract
};

export const allOperators = [
  add,
  allClear,
  clear,
  decimal,
  divide,
  divideAlt,
  equals,
  multiply,
  multiplyAlt,
  percent,
  positiveNegative,
  subtract
];

export const operators = [
  add,
  allClear,
  clear,
  divide,
  divideAlt,
  multiply,
  multiplyAlt,
  positiveNegative,
  subtract
];

export const controlLabels = {
  [allClear]: 'All clear',
  [clear]: 'Clear',
  [positiveNegative]: 'Positive/negative',
  [percent]: 'Percent',
  [multiplyAlt]: 'Multiply',
  [divideAlt]: 'Divide',
  [add]: 'Add',
  [subtract]: 'Subtract',
  [equals]: 'Equals',
  [decimal]: 'Decimal'
};

export const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];

export const keys = {
  backspace,
  enter
};

export const operatorKeys = [backspace, enter]
  .concat(...allOperators)
  .concat(numbers);
