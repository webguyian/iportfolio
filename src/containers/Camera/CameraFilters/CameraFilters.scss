.camera-filters {
  display: flex;
  flex-flow: column nowrap;
  transform: translateY(-5px);

  .camera-filter-list {
    display: flex;
    flex-flow: row nowrap;
    padding-bottom: $gutter;
    padding-left: 145px;
    overflow-x: auto;
    overflow-y: hidden;
  }

  .camera-filter {
    margin-left: $gutter;
    border: 2px solid transparent;

    &--active {
      border-color: $white;
    }

    &-label {
      margin: $gutter-s auto $gutter;
      font-size: $font-size-s;
      text-transform: uppercase;
    }
  }
}
