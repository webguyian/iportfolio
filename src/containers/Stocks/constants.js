export const initialStocks = [
  { symbol: 'AAPL', displaySymbol: 'AAPL', description: 'APPLE INC' },
  { symbol: 'AMZN', displaySymbol: 'AMZN', description: 'AMAZON.COM INC' },
  { symbol: 'FB', displaySymbol: 'FB', description: 'FACEBOOK INC-CLASS A' },
  { symbol: 'GOOG', displaySymbol: 'GOOG', description: 'ALPHABET INC-CL C' },
  { symbol: 'MSFT', displaySymbol: 'MSFT', description: 'MICROSOFT CORP' }
];

export const API_CANDLESTICK = 'https://finnhub.io/api/v1/stock/candle';

export const API_NEWS = 'https://finnhub.io/api/v1/news';

export const API_QUOTE = 'https://finnhub.io/api/v1/quote';

export const API_TOKEN = 'bojsah7rh5rcji5m4udg';

export const API_EXCHANGE = `https://finnhub.io/api/v1/stock/symbol?exchange=US&token=${API_TOKEN}`;
