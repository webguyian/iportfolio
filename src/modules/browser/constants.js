export const DEFAULT_COORDINATES = {
  latitude: 37.332,
  longitude: -122.03
};

export const API_TOKEN = `${process.env.API_PREFIX}/api/auth/token`;

export const MOCK_TOKEN = {
  expires: 1590934380,
  token: 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9'
};
