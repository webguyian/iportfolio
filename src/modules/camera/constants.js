export const defaultSettings = {
  flash: 'disabled',
  timer: 'disabled',
  filter: 'enabled',
  swap: 'disabled'
};

export const initialFilters = [
  {
    label: 'Original',
    value: 'none'
  },
  {
    label: 'Grayscale',
    value: 'grayscale(100%)'
  },
  {
    label: 'Sepia',
    value: 'sepia(100%)'
  },
  {
    label: 'Invert',
    value: 'invert(100%)'
  },
  {
    label: 'Saturate',
    value: 'saturate(250%)'
  },
  {
    label: 'High Contrast',
    value: 'contrast(400%)'
  },
  {
    label: 'Hue Rotate',
    value: 'hue-rotate(180deg)'
  },
  {
    label: 'Sharpen',
    value: 'url(#svgSharpen)'
  },
  {
    label: 'Blur',
    value: 'blur(4px)'
  },
  {
    label: 'Threshold',
    value: 'url(#svgThreshold)'
  }
];
