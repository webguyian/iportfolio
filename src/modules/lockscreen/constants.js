export const API_UNLOCK = `${process.env.API_PREFIX}/api/unlock`;
