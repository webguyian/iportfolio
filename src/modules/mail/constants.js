export const API_MAIL = `${process.env.API_PREFIX}/api/mail`;

export const defaultBody = '\n\nSent from iPortfolio';

export const initialValues = {
  to: 'hello@webguyian.com',
  from: '',
  subject: '',
  body: defaultBody
};
