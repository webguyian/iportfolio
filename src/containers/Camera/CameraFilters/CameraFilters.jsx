import React from 'react';
import { object, func, string } from 'prop-types';
import classNames from 'classnames';

import Button from 'components/Button/Button';
import Text from 'components/Text/Text';

import { getMobileOperatingSystem } from 'modules/browser/helpers';

if (getMobileOperatingSystem() === 'iOS') {
  require('context-filter-polyfill');
}

const CameraFilters = props => {
  const { activeFilter, filters, onFilterClick } = props;
  const baseClass = 'camera-filters';
  const filterClass = 'camera-filter';
  const activeClass = `${filterClass}--active`;
  const canvasClass = `${filterClass}-canvas`;

  return (
    <div className={baseClass}>
      <Text className={`${filterClass}-label`} type="display">
        {activeFilter}
      </Text>
      <svg xmlns="http://www.w3.org/2000/svg" style={{ display: 'none' }}>
        <filter id="svgThreshold">
          <feComponentTransfer>
            <feFuncR type="discrete" tableValues=" 0 1 1 1" />
            <feFuncG type="discrete" tableValues=" 0 1 1 1" />
            <feFuncB type="discrete" tableValues=" 0 1 1 1" />
          </feComponentTransfer>
        </filter>
        <filter id="svgSharpen">
          <feConvolveMatrix
            order="3 3"
            preserveAlpha="true"
            divisor="1"
            bias="0"
            kernelMatrix="-1,-1,-1 -1,9,-1 -1,-1,-1"
          />
        </filter>
      </svg>
      <div className={`${filterClass}-list`}>
        {filters.current.map(filter => (
          <Button
            className={classNames(
              filterClass,
              filter.label === activeFilter && activeClass
            )}
            key={filter.label}
            onClick={onFilterClick.bind(null, filter)}
            aria-label={filter.label}
          >
            <canvas
              className={canvasClass}
              ref={el => (filter.ref = el)}
              width={75}
              height={75}
            />
          </Button>
        ))}
      </div>
    </div>
  );
};

CameraFilters.propTypes = {
  activeFilter: string,
  filters: object.isRequired,
  onFilterClick: func.isRequired
};

export default CameraFilters;
