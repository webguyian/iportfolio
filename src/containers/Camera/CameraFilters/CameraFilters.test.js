import React from 'react';
import { render } from '@testing-library/react';

import { initialFilters } from 'modules/camera/constants';
import CameraFilters from './CameraFilters';

describe('<CameraFilters />', () => {
  const props = {
    activeFilter: 'Original',
    filters: { current: initialFilters },
    onFilterClick: jest.fn()
  };

  it('renders correctly', () => {
    const { asFragment } = render(<CameraFilters {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
