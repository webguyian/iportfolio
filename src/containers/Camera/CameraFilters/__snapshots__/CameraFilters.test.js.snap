// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CameraFilters /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="camera-filters"
  >
    <span
      class="ui-text ui-text--display camera-filter-label"
    >
      Original
    </span>
    <svg
      style="display: none;"
      xmlns="http://www.w3.org/2000/svg"
    >
      <filter
        id="svgThreshold"
      >
        <fecomponenttransfer>
          <fefuncr
            tableValues=" 0 1 1 1"
            type="discrete"
          />
          <fefuncg
            tableValues=" 0 1 1 1"
            type="discrete"
          />
          <fefuncb
            tableValues=" 0 1 1 1"
            type="discrete"
          />
        </fecomponenttransfer>
      </filter>
      <filter
        id="svgSharpen"
      >
        <feconvolvematrix
          bias="0"
          divisor="1"
          kernelMatrix="-1,-1,-1 -1,9,-1 -1,-1,-1"
          order="3 3"
          preserveAlpha="true"
        />
      </filter>
    </svg>
    <div
      class="camera-filter-list"
    >
      <button
        aria-label="Original"
        class="ui-btn camera-filter camera-filter--active"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Grayscale"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Sepia"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Invert"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Saturate"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="High Contrast"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Hue Rotate"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Sharpen"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Blur"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
      <button
        aria-label="Threshold"
        class="ui-btn camera-filter"
        type="button"
      >
        <canvas
          class="camera-filter-canvas"
          height="75"
          width="75"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
